### PR TITLE
docs: improve system events doc

### PR DESCRIPTION
**Pull request by TC-MO**
add admonitions
rework logic flow
rework structure & add headings
streamline descriptions within table

### CODE REVIEW

File: sources/platform/actors/development/programming_interface/system_events.md [hunk 1]
```diff
@@ -5,25 +5,40 @@ slug: /actors/development/programming-interface/system-events
...
-| `persistState` | `{ isMigrating: Boolean }` | Emitted in regular intervals (by default 60 seconds) to notify all components of the Apify SDK that it is time to persist their state in order to avoid repeating all work when the Actor restarts. This event is automatically emitted together with the migrating event, in which case the `isMigrating` flag is set to `true`. Otherwise, the flag is `false`. Note that the `persistState` event is provided merely for user convenience. You can achieve the same effect using `setInterval()` and listening for the `migrating` event. |
+| `cpuInfo`      | `{ isCpuOverloaded: Boolean }` | Emitted approximately every second, indicating whether the Actor is using maximum available CPU resources. |
+| `migrating`    | N/A | Signals that the Actor will soon migrate to another worker server on the Apify platform. |
+| `aborting`     | N/A | Triggered when a user initiaties a graceful abort of an Actor run, allowing time for cleanup. |
+| `persistState` | `{ isMigrating: Boolean }` | Emitted at regular intervals  (default: _60 seconds_) to notify Apify SDK componenets to persist their state. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/system_events.md:37]:
```suggestion
| `persistState` | `{ isMigrating: Boolean }` | Emitted at regular intervals  (default: _60 seconds_) to notify Apify SDK components to persist their state. |
```

Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/system_events.md:37]: We should probably link to a guide on configuring the interval (you can use the `APIFY_PERSIST_STATE_INTERVAL_MILLIS` environment variable).

File: sources/platform/actors/development/programming_interface/system_events.md [hunk 2]
```diff
@@ -5,25 +5,40 @@ slug: /actors/development/programming-interface/system-events
...
+# System events in Apify Actors
+
 **Learn about system events sent to your Actor and how to benefit from them.**
 
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
 ---
 
-The system notifies Actors about various events such as a migration to another server, [abort operation](#abort-another-actor) triggered by another Actor, or the CPU being overloaded.
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/system_events.md:15]: Why remove that link?

**Review by janbuchar — CHANGES_REQUESTED** [review]

File: sources/platform/actors/development/programming_interface/system_events.md (hunk 2 above)
Comment (TC-MO) [reply sources/platform/actors/development/programming_interface/system_events.md:15]: This is dead link not pointing to anything

**Pull request merged**